Documentation/contributor-guide/contributor-membership.md

```diff
@@ -0,0 +1,151 @@
+the codebase and software engineering principles. Their LGTM counts towards
+merging a code change into the project. A reviewer is generally on the ladder towards
+maintainership. 
+
+**Defined by:** *reviewers* entry in the [MAINTAINERS] file.
+
+### Requirements
+
+- member for at least 3 months
+- Primary reviewer for at least 5 PRs to the codebase
+- Reviewed or merged at least 20 substantial PRs to the codebase
+- Knowledgeable about the codebase
+- Sponsored by two active maintainer.
+    - Sponsors must be from multiple member companies to demonstrate integration across community.
+    - With no objections from other maintainers
+- Reviewers can be removed by a supermajority of the maintainers or can resign by notifying
+  the maintainers.
+
+### Responsibilities and privileges
+
+- Code reviewer status may be a precondition to accepting large code contributions
+- Responsible for project quality control via code reviews
+    - Focus on code quality and correctness, including testing and factoring
+    - May also review for more holistic issues, but not a requirement
+- Expected to be responsive to review requests
+- Assigned PRs to review related to area of expertise
+- Assigned test bugs related to area of expertise
+- Granted "triage access" to etcd project
+
+## Maintainers
+
+Maintainers are first and foremost contributors that have shown they
+are committed to the long term success of a project. Maintainership is about building
+trust with the current maintainers and being a person that they can
+depend on to make decisions in the best interest of the project in a consistent manner.
+
+**Defined by:** *maintainers* entry in the [MAINTAINERS] file.
+
+### Requirements
+
+- Deep understanding of the technical goals and direction of the project
+- Deep understanding of the technical domain of the project
+- Sustained contributions to design and direction by doing all of:
+    - Authoring and reviewing proposals
+    - Initiating, contributing and resolving discussions (emails, GitHub issues, meetings)
+    - Identifying subtle or complex issues in designs and implementation PRs
+- Directly contributed to the project through implementation and / or review
+- Subscribed to [etcd-maintainers@googlegroups.com]
+- Elected by supermajority of active maintainers.
+
+### Responsibilities and privileges
+
+- Make and approve technical design decisions
+- Set technical direction and priorities
+- Define milestones and releases
+- Mentor and guide reviewers, and contributors to the project.
+- Participate when called upon in the [security disclosure and release process]
+- Ensure continued health of the project
+    - Adequate test coverage to confidently release
+    - Tests are passing reliably (i.e. not flaky) and are fixed when they fail
+- Ensure a healthy process for discussion and decision making is in place.
+- Work with other maintainers to maintain the project's overall health and success holistically
+
+### Retiring
```

Comment:
I think the retiring process should be applied to all roles, not just Maintainers.

Reply:
We may also want to clearly request all roles (Members, Reviewers and Maintainers) to notify the community when they are going to be inactive for more than 1 week, e.g. on vacation without computer.

Reply:
I wouldn't require this from Members definitely. I'm not sure about reviewers but probably not. 
That make sense for maintainers, as it's likely blocking for others.

Reply:
Agree with @ptabor, I'm also not super pressed to clarify process of removing members/reviewers, as we first need to nominate them :P. Let's do it in followup.